lib/plugins/aws/package/compile/functions/index.js

```diff
@@ -469,8 +469,11 @@ class AwsCompileFunctions {
           // Include function configuration in version id hash (without the Code part)
           const properties = _.omit(_.get(functionResource, 'Properties', {}), 'Code');
           _.forOwn(properties, value => {
-            const hashedValue = _.isObject(value) ? JSON.stringify(value) : _.toString(value);
-            versionHash.write(hashedValue);
+            if (_.isObject(value)) {
+              versionHash.write(JSON.stringify(value));
+            } else {
+              versionHash.write(value ? String(value) : '');
```

Comment:
Let's put condition as `value != null ?` (otherwise we're ignoring `0` and other falsies, which may intentionally be represented)

Reply:
In lodash
```
_.isObject(null);
// => false
```
So condition ```versionHash.write(value ? String(value) : '')``` checked like condition ```value != null ?```

Reply:
See my comment in other discussion